src/rpc/client.zig

```diff
@@ -1,2338 +1,679 @@
-            defer rangeObj.deinit();
-            try rangeObj.put("firstSlot", .{ .integer = @as(i64, @intCast(range.firstSlot)) });
-            if (range.lastSlot) |lastSlot| {
-                try rangeObj.put("lastSlot", .{ .integer = @as(i64, @intCast(lastSlot)) });
-            }
-        }
-
-        var arrList = std.ArrayList(json.Value).init(self.allocator);
-        defer arrList.deinit();
-        try arrList.append(.{ .object = optionsObj });
-
-        const params = .{ .array = arrList };
-
-        var tree = try self.makeRequestWithJsonValueResponse(json.Value, "getBlockProduction", "1", params);
-        const errorObject = tree.root.object.get("error");
-        const id = tree.root.object.get("id").?.string;
-
-        var arena = std.heap.ArenaAllocator.init(self.allocator);
-        const responseAllocator = arena.allocator();
-        if (errorObject) |errObj| {
-            return jsonrpc.ResponseAlt(types.BlockProductionInfo).init(arena, Result{ .jsonrpc = "2.0", .id = id, .result = null, .@"error" = jsonrpc.ErrorObject{
-                .code = errObj.object.get("code").?.integer,
-                .message = errObj.object.get("message").?.string,
-            } }, tree);
-        }
-
-        var result = tree.root.object.get("result").?;
-        var context = result.object.get("context").?;
-        var value = result.object.get("value").?;
-
-        var byIdentity = std.StringArrayHashMap([]u64).init(responseAllocator);
-        var iter = value.object.get("byIdentity").?.object.iterator();
-
-        while (iter.next()) |entry| {
-            var vals = std.ArrayList(u64).init(responseAllocator);
-            for (entry.value_ptr.*.array.items) |val| {
-                try vals.append(@as(u64, @intCast(val.integer)));
-            }
-            try byIdentity.put(entry.key_ptr.*, vals.items);
-        }
-
-        return jsonrpc.ResponseAlt(types.BlockProductionInfo).init(
-            arena,
-            Result{ .jsonrpc = "2.0", .id = id, .result = .{ .context = .{
-                .apiVersion = context.object.get("apiVersion").?.string,
-                .slot = @as(u64, @intCast(context.object.get("slot").?.integer)),
-            }, .value = .{ .byIdentity = byIdentity, .range = .{
-                .firstSlot = @as(u64, @intCast(value.object.get("range").?.object.get("firstSlot").?.integer)),
-                .lastSlot = @as(u64, @intCast(value.object.get("range").?.object.get("firstSlot").?.integer)),
-            } } } },
-            tree,
-        );
-    }
-
-    pub fn getBlockCommitment(self: *Self, slot: u64) !jsonrpc.Response(types.BlockCommitment) {
-        var params = [1]u64{slot};
-        return try self.makeRequestWithJRpcResponse(types.BlockCommitment, []u64, "getBlockCommitment", "1", &params);
+    pub fn getAccountInfo(self: *Client, arena: *std.heap.ArenaAllocator, pubkey: Pubkey, config: GetAccountInfoConfig) !AccountInfo {
```

Comment:
Originally I went with this approach because I found it more explicit since an arena should be used when parsing using std.json as it does no track allocations. I think on reflection though it would probably be better to take a std.mem.Allocator and return a std.json.Parsed(T) (T wrapped with the arena used for parsing) because it will be cleaner on the user side

Reply:
would you want to tackle this in another pr?

Reply:
Yeah thats what I was thinking. I'll add docs to my list of todo's for the next PR as well.